src/rendering/line_iter.rs

```diff
@@ -321,6 +324,15 @@ where
                 }
 
                 Token::Word(w) => {
+                    if self.empty {
+                        // If this is the first word on the line, offset the line by
+                        // the word's left negative boundary to make sure it is not clipped.
+                        let offset = handler.measure_left_offset(w);
```

Comment:
Can we somehow avoid double-measuring here? I'd probably prefer modifying `measure` to return the offset along with the width.

Reply:
Done! Realised that having a more specialised function was nicer than changing the existing `measure` function, to avoid touching a lot of places where it is used.